pkgs/applications/audio/virtual-ans/default.nix

```diff
@@ -0,0 +1,54 @@
+{ stdenv
+, fetchurl
+, unzip
+, xorg
+, libGL
+, alsaLib
+, SDL2
+, autoPatchelfHook
+}:
+
+stdenv.mkDerivation rec {
+  name = "virtual-ans-${version}";
+  version = "3.0.2c";
+
+  src = fetchurl {
+    url = "https://warmplace.ru/soft/ans/virtual_ans-${version}.zip";
+    sha256 = "1dd4d91yjmp7xfjcjlam8l55r4963ypjgll97n4gjvl72m1fq856";
+  };
+
+  meta = with stdenv.lib; {
+    description = "Photoelectronic microtonal/spectral musical instrument";
+    homepage = "https://warmplace.ru/soft/ans/";
+    license = licenses.free;
+    platforms = platforms.linux;
+    maintainers = with maintainers; [ jacg ];
+  };
+
+  nativeBuildInputs = [
+    unzip
+    autoPatchelfHook
+    stdenv.cc.cc.lib
+    xorg.libX11
+    xorg.libXi
+    libGL
+    alsaLib
+    SDL2
+  ];
+
+  unpackPhase = ''
+    unzip $src
+  '';
+
+  installPhase = let unzipdir = "virtual_ans3"; in ''
+      mkdir -p $out
+      cp -R ${unzipdir}/* $out/
+      ln -s $out/${startScript} $out/bin/virtual-ans
+  '';
+
+  startScript = if stdenv.isx86_32 then "START_LINUX_X86"
```

Comment:
Is using their startup script here required? From what i can tell that essentially just launches a binary, which we can also just do without that script.

Reply:
The startup script

1. Unwraps symlinks

2. Picks a platform-appropriate binary

3. Passes `boot.pixcode` as an argument to the binary.

I'm not sure what we gain by not using the script. Even if the symlink unwrapping turns out to be unnecessary, we'd still need to reproduce picking the binary and passing the boot code to it. And I have no idea what has to be done for Darwin (in case someone wants get that to work). So it seems simplest just to use the script provided for whatever architecture we happen to be on.

Reply:
Fair enough :+1: